src/input/keyboard/mod.rs

```diff
@@ -556,6 +556,24 @@ impl<D: SeatHandler + 'static> KeyboardHandle<D> {
         );
     }
 
+    /// Iterate over the keysyms of the currently pressed keys.
+    pub fn with_pressed_keysyms<F>(&self, mut f: F)
+    where
+        F: FnMut(KeysymHandle<'_>),
```

Comment:
Would it be more useful to have the closure provided an iterator rather than running multiple times?

Reply:
Not sure, I just followed the other apis. Due to borrowing state behind a Mutex for the KeysymHandle, we sadly can't simply return an Iterator...

I suppose this is a more general question about how we want to approach the few `with_` functions, that we have?

Reply:
ping @i509VCB